src/BuildScriptGenerator/DotNetCore/DefaultAspNetCoreWebAppProjectFileProvider.cs

```diff
@@ -81,15 +84,29 @@ public string GetProjectFile(ISourceRepo sourceRepo)
             // to web sdk.
             if (projectFile == null)
             {
-                var projectFiles = sourceRepo
-                    .EnumerateFiles($"*.{DotnetCoreConstants.ProjectFileExtensionName}", searchSubDirectories: true);
+                // search for .csproj files
+                var projectFiles = GetAllProjectFilesInRepo(
+                        sourceRepo,
+                        DotnetCoreConstants.CSharpProjectFileExtensionName);
 
                 if (!projectFiles.Any())
```

Comment:
I'm not sure I understand the idea here - the root is searched first, and if nothing is found, then **the root and its subdirs** are searched as well.
Maybe these two steps could be coalesced into one "search root AND subdirs" step?

Reply:
This is for minor perf improvement (atleast currently AppService recommends using it like this, but in future they need not) to avoid unnecessary lookup into the sub-directories.